dascore/proc/filter.py

```diff
@@ -265,6 +249,80 @@ def median_filter(
     return patch.update(data=new_data)
 
 
+@patch_function()
+def notch_filter(patch: PatchType, q, **kwargs) -> PatchType:
+    """
+    Apply a second-order IIR notch digital filter on patch's data.
+
+    A notch filter is a band-stop filter with a narrow bandwidth (high quality factor).
+    It rejects a narrow frequency band and leaves the rest of the spectrum
+    little changed.
+
+    Parameters
+    ----------
+    patch
+        The patch to filter
+    q
+        Quality factor (float). A higher Q value means a narrower notch,
+        which targets the specific frequency more precisely.
+        A lower Q results in a wider notch, meaning a broader range of
+        frequencies around the specific frequency will be attenuated.
+    **kwargs
+        Used to specify the dimension(s) and associated frequency and/or wavelength
+        (or equivalent values) for the filter.
+
+    Notes
+    -----
+    See [scipy.signal.iirnotch]
+        (https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.iirnotch.html)
+        for more information.
+
+    Examples
+    --------
+    >>> import dascore
+    >>> pa = dascore.get_example_patch()
+
+    >>>  # Apply a notch filter along time axis to remove 60 Hz
+    >>> filtered = pa.notch_filter(time=60, q=30)
+
+    >>>  # Apply a notch filter along distance axis to remove 0.2 m wavelength
+    >>> filtered = pa.notch_filter(distance=0.2, q=10)
+
+    >>>  # Apply a notch filter along both time and distance axes
+    >>> filtered = pa.notch_filter(time=60, distance=0.2, q=40)
+
+    >>> # Optionally, units can be specified for a more expressive API.
+    >>> from dascore.units import m, ft, s, Hz
+    >>> # Apply a notch filter along time axis to remove 60 Hz
+    >>> filtered = pa.notch_filter(time=60 * Hz, q=30)
+    >>> # Apply a notch filter along distance axis to remove 0.2 m wavelength
+    >>> filtered = pa.notch_filter(distance=0.2 * m, q=30)
+    """
+    data = patch.data
+    for coord_name, info in get_multiple_dim_value_from_kwargs(patch, kwargs).items():
+        dim, ax, value = info["dim"], info["axis"], info["value"]
+        coord = patch.get_coord(coord_name)
+
+        inverted_units = True
+        # Invert units if needed
+        if isinstance(value, dc.units.Quantity) and coord.units is not None:
+            value, inverted_units = get_inverted_quant(value, coord.units)
+
+        # Check valid parameters
+        if inverted_units:
+            w0 = to_float(value)
+        else:
+            w0 = to_float(1 / value)
```

Comment:
hmmm... its a bit odd this is required. Doesn't `get_inverted_quant` already do this?

Reply:
Specifically [here](https://github.com/DASDAE/dascore/blob/3d635c1333a106ec5b48a65a7f463e8424cbf190/dascore/units.py#L266-L267).

Reply:
Thanks for looking into this more closely. I did not understand the behavior of the `get_inverted_quant` function well. We actually don't need this. Please see my last commit (changed examples and a test) for further details.